drivers/bluetooth/hci/userchan.c

```diff
@@ -257,14 +291,17 @@ SYS_INIT(bt_uc_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
 
 static void cmd_bt_dev_found(char *argv, int offset)
 {
-	if (strncmp(&argv[offset], "hci", 3) || strlen(&argv[offset]) < 4) {
-		posix_print_error_and_exit("Error: Invalid Bluetooth device "
-					   "name '%s' (should be e.g. hci0)\n",
-					   &argv[offset]);
-		return;
+	arg_found = true;
+	if (strncmp(&argv[offset], "hci", 3) == 0 && strlen(&argv[offset]) >= 4) {
+		/* It's a hciX string */
+		hci_socket = true;
+		bt_dev_index = strtol(&argv[offset + 3], NULL, 10);
+	} else if (sscanf(&argv[offset], "%15[^:]:%d", ip_addr, &port) == 2) {
+		hci_socket = false;
+	} else {
+		posix_print_error_and_exit("Invalid option for --bt-dev. "
```

Comment:
Very minor: It would be nice printing out the option we don't like
(similar to how it was before:
posix_print_error_and_exit("Invalid option %s ..", argv);

Reply:
Seems more informative, will add this